### DISCUSSION TITLE
Reconstructing a Varian CBCT

### DISCUSSION THREAD

**Discussion by scoots38** [CERN/TIGRE#568] [Q&A]
Hi Ander!

Thank you so much for this amazing toolbox! There is a lot of really cool stuff in here. I am trying to reconstruct a CBCT from a Varian machine. I have run into a few issues. I am using the MATLAB codes in TIGRE.

First off I want to make sure that I am using the functions in Utilities\IO\VarianCBCT properly. First I am using GeometryFromXML, ProjLoader, and BlkLoader to upload the projections, geometry, etc. Then I use LogNormal to convert the projections to attenuation values. Then, I use ScCalibFromXML to get the necessary info to run the ScatterCorrection function to reduce scatter in the projections. Then I pick a recon algorithm and get a CT image. Is that a good sequence to get nice images in the end? Am I missing anything? I saw that there are Beam Hardening correction functions as well but I haven't been able to figure out how to use those yet. The BHCorrection and BH_Remapping functions do not come with an in-depth description of how to use them. And some the output variables are named the same as some of the input variables.

For some reason, with the specific scan I am working with, the Gradient-based Algorithms and MLEM do not give very good results. The image is super blurry or has bad ringing artifacts. But, the Krylov subspace algorithms give pretty good results. Do you have any idea why that could be? Here is a reconstruction with IRN_TV_CGLS (not bad, but could be better):

![image](https://github.com/user-attachments/assets/49466df7-44bb-43d8-82c6-9eb2fef8f14f)

Here is TrueBeam's reconstruction (I can't get it to be this good):
![image](https://github.com/user-attachments/assets/7ce690ae-f63c-4923-b90f-731a07201545)

Any idea how I can rival the reconstruction that TrueBeam provides? Do you have any general tips on working with Varian projections?

As I have been reading through papers and the README on TIGRE I found that ASD_POCS is good for an image with limited projections. Is there are more extensive description of what algorithms work best under certain situations? There are a lot to choose from and it sometimes feels like I am just guessing which one to use.

Sorry for the long message with so many questions! I hope they were clear.

Scott

**Comment by AnderBiguri** [comment]
Hi @scoots38 ,
I don't have much hands-on experience with the Varian data loader, but the idea of the functions is that you don't need to call them one by one yourself, instead just do `[proj,geo,angles] = VarianDataLoader(filename)` and the function will take care of everything. Have you tried that?

About recon: my suggestion is that if you are unsure if the code is correct, always use FDK or OS-SART to test. Once you are happy with that, indeed explore. The reality is that some algorithms are more fiddly to set up correctly than others, and indeed `IRN_TV_CGLS` is one that under the very right choice of parameters produces a very nice result, but not otherwise. In any case, TrueBeam should be likely doing FDK.

But indeed, its hard to know which algorithm to use. I tend to suggest what I put here: https://github.com/CERN/TIGRE/tree/master/MATLAB/Algorithms#readme

But do always make sure you get a reasonable image with FDK/OS-SART, as in theory, our VarianDataLoader was compared and tested against Varian recon like that. Perhaps @yidu-bjcancer can give more insight.

**Reply by scoots38** [reply]
I did not realize that VarianDataLoader did everything for you! My images already look much better. Thanks for showing me that. Just as a heads up, the Beam Hardening functions don't seem to work properly. It could be that my Varian data doesn't have the necessary calibration factors for the functions to work, but it might be a bug. I don't think I will need that to work anytime soon, I just thought I would point it out.

And thanks for the other tips!

**Reply by AnderBiguri** [reply]
Great! Glad that it helped :) Do let me know if you investigate more in the BeamHardening side. Unforutnately I currently don't have access to a Varian machine so I have no much way to improve that without help, so do ping me if that is something you want to investigate at some point.
